contrib/masterfiles-stage/options.sh

```diff
@@ -1,18 +1,114 @@
-# Nice option parsing with nk412/optparse
-# Source the optparse.bash file ---------------------------------------------------
-source "${DIRNAME}/lib/optparse/optparse.bash"
-# Define options
-optparse.define short=d long=deploy-dir desc="The directory where masterfiles should be deployed after successful staging" variable=opt_deploy_dir  default=/var/cfengine/masterfiles
-optparse.define short=p long=params-file desc="The file that contains the paramaters necessary to interface with UPSTREAM" variable=opt_params_file default=/opt/cfengine/dc-scripts/params.sh
-optparse.define short=v long=verbose desc="Flag to set verbose mode on" variable=verbose_mode value=true default=false
-# Because it's currently in active development, we are defaulting debug mode to on
-optparse.define short=D long=DEBUG desc="DEBUG mode and do not log to the logfile" variable=debug_mode value=true default=false
-# Source the output file ----------------------------------------------------------
-source $( optparse.build )
+# Original option parsing code generated thanks to
+# the bash option parsing library at nk412/optparse on github.
+#
+# Generated code modified and refined by Mike Weilgart <mikeweilgart@gmail.com>
+
+usage() {
+cat << EOF
+usage: $0 [OPTIONS]
+
+OPTIONS:
+
+        -d --deploy-dir [directory]
+                            The directory where masterfiles should be
+                            deployed after successful staging.
+                            Default: /var/cfengine/masterfiles
+
+        -p --params-file [filename]
+                            The file that contains the parameters necessary
+                            to interface with UPSTREAM.
+                            Default: /opt/cfengine/dc-scripts/params.sh
+
+        -v --verbose:       Set verbose mode on
+        -D --DEBUG:         DEBUG mode (do not log to the logfile)
+        -? --help:          Display this usage message
+
+EOF
+}
+
+# Contract long options into short options
+params=()
+for param; do
+  case "$param" in
+    --deploy-dir)
+      params+=("-d")
+      ;;
+    --params-file)
+      params+=("-p")
+      ;;
+    --verbose)
+      params+=("-v")
+      ;;
+    --DEBUG)
+      params+=("-D")
+      ;;
+    "-?"|--help)
+      usage
+      exit 0
+      ;;
+    --*)
+      printf '%s\n' "Unrecognized long option: $param"
+      usage
+      exit 2
+      ;;
+    *)
+      # Assume that any other args should stay as-is
+      # (This would include short option flags as well
+      # as any other arguments passed to the script.)
+      params+=("$param")
+      ;;
+  esac
+done
+
+set -- "${params[@]}"
+
+# Set default variable values
+# (Don't change these without updating usage.)
+MASTERDIR=/var/cfengine/masterfiles
```

Comment:
@nickanderson, considering we're putting everything else in the params file—path to staging directory (or ROOT is what we use now), channel config file path (to be changed to a bash array directly in `params.sh`)—what do you think about removing the `-d` option and requiring any custom deployment dir to be set directly in `params.sh`?  (We would keep the default `MASTERDIR` value for backward compatibility.)

This would get us even closer to fully drop-in replacement, as then you could specify a custom `MASTERDIR` in your custom `params.sh` file—and put that `params.sh` file in the default location to be read in automatically.